sdk/python/tests/dsl/component_bridge_tests.py

```diff
@@ -194,3 +194,21 @@ def consume_list(list_param: list) -> int:
         full_command_line = task.command + task.arguments
         for arg in full_command_line:
             self.assertNotIn('PipelineParam', arg)
+
+    def test_converted_outputs(self):
+        component_text = textwrap.dedent('''\
+            outputs:
+            - name: Output 1
+            implementation:
+                container:
+                    image: busybox
+                    command:
+                    - producer
+                    - {outputPath: Output 1}  # Outputs must be used in the implementation
```

Comment:
Sorry perhaps a naive question: what if a user specify two output paths that cause potential name collision in the component spec? For example:
```
command:
  - {outputPath: Output 1}
  - {outputPath: output_1}
```

Reply:
This is not a naive question at all =)
I care about such things, so I've created the `generate_unique_name_conversion_table` function. It takes a collection of strings, applies a conversion function to each item, but adds indexes if the converted names start conflicting.
So in your example, previously there would be two outputs: `["output_1", "output_1_2"]` and after the change there would be three: `["output_1", "output_1_2", "Output_1"]`. I should add a test for that.

This PR just moved the call of that function (from L45 to L70), but it's still called, so the conversion table should remain the same as before.

P.S. But your question got me thinking harder and I think I've discovered an existing issue (renaming was not stable for python 3.5.) and an edge case (when index could be added to a name that is unchanged by the conversion function). I'll think of fixing them in a different PR. However it might be unfeasible, since the issues were present in the old code too.

Reply:
I tested locally by adding `{outputPath: output_1}`, and `load_component_from_text` raises
```
TypeError: Argument "OutputPathPlaceholder(output_name='output_1')" references non-existing output.
```
Can you reproduce this locally?

Reply:
> I tested locally by adding `{outputPath: output_1}`, and `load_component_from_text` raises
> 
> ```
> TypeError: Argument "OutputPathPlaceholder(output_name='output_1')" references non-existing output.
> ```
> 
> Can you reproduce this locally?

Did you also add that output to the `outputs:` section?

Maybe you're a bit confused about the scope of this PR changes:

The component.yaml and component library itself has always supported arbitrary input and output names and compared them **exactly** without any sort of conversion. This PR does not change that.

The conversion was only happening when converting to `ContainerOp` in the KFP dsl bridge. Now this conversion is no longer necessary as `PipelineParam` supports arbitrary name. This PR adds unconverted output names to `ContainerOp.outputs` while keeping the converted names available as well.

Reply:
Oh I see. That makes sense. Thanks

Reply:
> I tested locally

I really love that you're eager to try and test the new things =)
I appreciate that.